tests/serviceBuilder.test.js

```diff
@@ -1146,5 +1149,240 @@ tap.test('serviceBuilder', test => {
     assert.end()
   })
 
+  test.test('timeout', async assert => {
+    assert.test('returnAs: JSON', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
+        .replyContentLength()
+        .get('/foo')
+        .delay(101)
+        .reply(200, { the: 'response' }, {
+          some: 'response-header',
+        })
+
+      const service = serviceBuilder('my-service-name')
+
+      await assert.rejects(async() => {
+        await service.get('/foo', {}, {
+          returnAs: 'JSON',
+          timeout: 100,
+        })
+      }, new Error('Request timed out'))
+
+      myServiceNameScope.done()
+      assert.end()
+    })
+
+    assert.test('returnAs: BUFFER', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
+        .replyContentLength()
+        .get('/foo')
+        .delay(101)
+        .reply(200, { the: 'response' }, {
+          some: 'response-header',
+        })
+
+      const service = serviceBuilder('my-service-name')
+
+      await assert.rejects(async() => {
+        await service.get('/foo', {}, {
+          returnAs: 'BUFFER',
+          timeout: 100,
+        })
+      }, new Error('Request timed out'))
+
+      myServiceNameScope.done()
+      assert.end()
+    })
+
+    assert.test('returnAs: STREAM with delay body', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
+        .replyContentLength()
+        .get('/foo')
+        .delayBody(101)
+        .reply(200, { the: 'response' }, {
+          some: 'response-header',
+        })
+
+      const service = serviceBuilder('my-service-name')
+
+      const response = await service.get('/foo', {}, { returnAs: 'STREAM', timeout: 100 })
+      assert.equal(response.statusCode, 200)
```

Comment:
if delay > timeout why this request is resolved successfully?

Reply:
Here I delayed only the body, timeout is on connection delay (it is the difference from [delay and delayBody](https://github.com/nock/nock#delay-the-response)).
In this way, I test that a stream response does not interrupt also with the timeout response.